Morbid surgery expansion

plsleavemealon opened the pull request:
## About The Pull Request


This PR tweaks how tools with the CRUEL_IMPLEMENT interact with patients. Essentially, when performing a surgery without anesthetic, it will give them the same mood debuff that they would suffer if you had failed a surgery step (fairly long lasting mood debuff that is removeable with alcohol or a nap).

Also adds experimental surgeries to morbid surgeries, meaning coroners now do them faster and get a mood bonus for completing them. The experimental surgery step now also causes a lot of pain when performed by someone who is morbidly inclined, such as the coroner. This step is the part where you are interacting with the patient with an open hand and is contextual to the experimental surgery (thread the patients veins, ground their nerves, etc). Anesthetic will mitigate this entirely.

## Why It's Good For The Game

Cruel tools are cruel for a reason, using them on the living is akin to torture. So they are going to hurt, a lot. And this gives those tools some added mechanical flavor befitting of a mad doctor. Experimental surgeries are all quite morbid in general, so its very thematic to add them to the list of morbid surgeries. And the part where you start interacting with the body is perhaps the most morbid part of those surgeries! It also makes sense for that to cause quite a lot of pain, its a delicate procedure so you really should have the patient under anesthesia of some kind.

## Changelog

:cl:
add:experimental surgeries are now morbid surgeries, and as such coroners, or anyone else morbidly inclined, get a slight speed boost when performing them. Coroner's, or anyone morbidly inclined will also get a mood boost from completing these surgeries.
add:morbid tools now cause a great deal of pain for the patient if used without anesthesia. The experimental surgery step will now cause a great deal of pain when performed by someone who is morbidly inclined, such as the coroner, however this can be avoided with anesthesia.
/:cl:

plsleavemealon committed:
testing

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon committed:
Update objects.dmi

plsleavemealon committed:
Merge branch 'master' into morbid-surgery-expansion

necromanceranne reviewed (dismissed):
I'm unfortunately not happy with any of these changes.

> Third, when using morbid tools upon a living patient that is conscious and not under anesthetic, it will give a hefty mood penalty to the patient. 

This is actually the opposite of the flavour. The punishment for this is already on the _patient_, not the surgeon. And the surgeon, in question, is kind of a freak, and I want it to be dubious as to whether or not they even care to give you anesthetic. I think that's more interesting. 

Also I listened to Night Surgeon on repeat while coding Morbid, so make of that what you will.

> Second, it expands the bonus that morbid tools grant to work for all surgeries if performed on a dead target. Morbid surgeries that are performed on a living target still grant the speed bonus as normal. So a user with the morbid trait that is operating on a corpse will get a speed bonus, regardless of the operation being performed.

This also is a a problem. Primarily because it turns the coroner into an extremely efficient medical doctor in the event that someone is dead. And that's quite a few patients. One of the things I wanted to avoid with adding Morbid was making them, by default, better than medical doctors during corpse recovery. For that, I wanted the two to be on largely equal footing, but incentivize the coroner to start with an autopsy so that they can get a speed boost (and give the dead guy a permanent buff, so a win for everyone). They're still expected to do coroner things even when they're acting as a medical doctor.

The Morbid surgeries exist because they're flavourful and subtly encourage the coroner to seek people out to perform them on people. If this was extended to, say, the experimental surgeries, this wouldn't be so much of a problem. But if it's literally tend wounds, thats when we have hit a snag, because that's the most commonly performed surgery in the game and also the bread and butter of medical doctors. I don't want to see coroners taking away their spotlight in that area.

> First it causes morbid tools (any tool with the CRUEL_IMPLEMENT tag) to work slower when used on living targets for non-morbid surgeries, specifically, roughly 30% slower.

I think if you wanted to further disincentivize coroners doing normal surgeries on people, this would be preferable on its own. But I would think this should be on the Morbid person, and not the tools. I would rather the coroner not have to feel like they need to go get regular tools to not suffer a speed penalty. And be specifically for tend wounds and organ repair (except brain surgery) on the living. It's another flavour issue. The problem is the person, not the tools, for why these penalties exist.

Your changelog is also not very descriptive. This is something that is player facing, so try to elaborate on what changed succinctly for the sake of someone trying to understand what you changed. Just telling them that interactions have expanded doesn't tell them very much.

plsleavemealon commented:
> This is actually the opposite of the flavour. The punishment for this is already on the _patient_, not the surgeon. And the surgeon, in question, is kind of a freak, and I want it to be dubious as to whether or not they even care to give you anesthetic. I think that's more interesting.
> 

The reason behind this is is that if they are under analgesic, they shouldn't feel pain. It would still be a creepy experience. This was also the easiest way I could think of to implement a harsh penalty for the patient. Its not so much that the doctor is suffering a penalty from it, more along the lines of the patient begging you to use anesthetic because good lord IT HURTS.


> This also is a a problem. Primarily because it turns the coroner into an extremely efficient medical doctor in the event that someone is dead. And that's quite a few patients. One of the things I wanted to avoid with adding Morbid was making them, by default, better than medical doctors during corpse recovery. For that, I wanted the two to be on largely equal footing, but incentivize the coroner to start with an autopsy so that they can get a speed boost (and give the dead guy a permanent buff, so a win for everyone). They're still expected to do coroner things even when they're acting as a medical doctor.
> 
> The Morbid surgeries exist because they're flavourful and subtly encourage the coroner to seek people out to perform them on people. If this was extended to, say, the experimental surgeries, this wouldn't be so much of a problem. But if it's literally tend wounds, thats when we have hit a snag, because that's the most commonly performed surgery in the game and also the bread and butter of medical doctors. I don't want to see coroners taking away their spotlight in that area.
> 

I figured letting them be more efficient at handling corpses would help further incentivize people to play coroner, and also further encourage doctors to let the coroner work with the corpses. Oftentimes the coroner struggles at getting the doctors to just...let them work when corpses come in. The other thing is, there is only 1 coroner, there are usually multiple doctors, so the coroner being a bit more efficient at tending the corpses wouldn't be much of an issue. At least thats what I feel.

> 
> I think if you wanted to further disincentivize coroners doing normal surgeries on people, this would be preferable on its own. But I would think this should be on the Morbid person, and not the tools. I would rather the coroner not have to feel like they need to go get regular tools to not suffer a speed penalty. And be specifically for tend wounds and organ repair (except brain surgery) on the living. It's another flavour issue. The problem is the person, not the tools, for why these penalties exist.
> 
> Your changelog is also not very descriptive. This is something that is player facing, so try to elaborate on what changed succinctly for the sake of someone trying to understand what you changed. Just telling them that interactions have expanded doesn't tell them very much.

The speed penalty wasn't so much to dissuade the coroner from using these on the living, but more to dissuade doctors from doing so. If anything, the speed malus is a **boon** to the coroner, as they are likely using these specifically to draw out the patients pain and suffering while they tend their injuries.

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon committed:
tweaking cruel tool penalty trigger

plsleavemealon commented:
I went ahead and tweaked the cruel tool penalty, now if used by someone who lacks the coroner's morbid trait they will operate slower on a living patient, but they will act like normal tools when used on a cadaver. Coroner's will operate on them as if they were normal tools (though the patient is going to suffer that immense mood penalty that I mentioned before).

plsleavemealon committed:
more tweaking of the cruel tool penalty on the living

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon commented:
@necromanceranne thoughts on my reasoning?

necromanceranne commented:
> The way I have it setup, the patient also gets the special mood thoughts for the surgery failure if they can feel the pain from it (not under anesthetic). Its literally the same moodlets that you'd suffer from surgery failure. I should also clarify...the moodlet is on the patient. Not on the person operating. The patient is having an awful time. But the madman using the tools? Probably amused if anything. Unless they are using the tools out of desperation

In that case I read that very wrong. My bad. I had just woken up.

> I figured letting them be more efficient at handling corpses would help further incentivize people to play coroner, and also further encourage doctors to let the coroner work with the corpses. Oftentimes the coroner struggles at getting the doctors to just...let them work when corpses come in. The other thing is, there is only 1 coroner, there are usually multiple doctors, so the coroner being a bit more efficient at tending the corpses wouldn't be much of an issue. At least thats what I feel. EDIT: Also...a decent chunk of players view the coroner as an "autopsy bot." With their only job/service being....autopsies. And that feels bad to me.

That is ostensibly what the role is. They do autopsies. While I do get that people are more eager to get bodies up and moving than entertain the coroner's shtick, that conflict is more or less on purpose to a degree. They need to go participate in medical a bit to get people to let them take the bodies.

> The speed penalty wasn't so much to dissuade the coroner from using these on the living, but more to dissuade doctors from doing so. If anything, the speed malus is a **boon** to the coroner, as they are likely using these specifically to draw out the patients pain and suffering while they tend their injuries. And the longer the surgery, the more pain and suffering it causes.

Still, that is counterproductive to the coroner having specializations in certain surgeries. Which can be done while alive. There is flavour and then there is practicality for accomplishing that flavour. 

As for people stealing the tools, I'm conscious of that, but seemingly that is the intent. Or something. I don't agree with it, but it is what it is.

> EDIT: Is this changelog more clear?

Maybe explain that they have to be under to avoid the mood penalty? And your last line repeats itself at the end.

plsleavemealon commented:
> 
> That is ostensibly what the role is. They do autopsies. While I do get that people are more eager to get bodies up and moving than entertain the coroner's shtick, that conflict is more or less on purpose to a degree. They need to go participate in medical a bit to get people to let them take the bodies.
> 
The issue is this leads to negative friction, where the doctor just ends up arguing with the coroner who is trying to do their job, which gets extra funny when there are living but critical patients present.

I view the coroner as focused on the dead. Any operations they do on the living is a sideshow to the main course. So the amount of patients they get are automatically more limited then a doctor because of this. Doctors operate on both living and dead, Coroner's focus on the dead.

So I always viewed the workflow like this:

Alive? Doctor. If no doctor, all doctors busy, or emergency then any other medical personnel (including coroner) can take care of it.
Dead? Coroner. If no coroner, or the coroner is busy, or emergency, then doctor. If both doctor and coroner are not present or busy, then any other medical personnel can handle it as they see fit.

> > The speed penalty wasn't so much to dissuade the coroner from using these on the living, but more to dissuade doctors from doing so. If anything, the speed malus is a **boon** to the coroner, as they are likely using these specifically to draw out the patients pain and suffering while they tend their injuries. And the longer the surgery, the more pain and suffering it causes.
> 
> Still, that is counterproductive to the coroner having specializations in certain surgeries. Which can be done (and practically speaking, probably would be done) while alive. There is flavour and then there is practicality for accomplishing that flavour. I think the mood penalty from earlier might be enough.
> 
> As for people stealing the tools, I'm conscious of that, but seemingly that is the intent. Or something. I don't agree with it, but it is what it is.
> 

Its not so much them stealing the tools, its to give incentives to avoid using them outside of niche purposes.

A doctor might use these tools because all the other tools are for whatever reason, in use and they have no others left. The patient is not going to be pleased with this, especially if they lack anesthetic as they are going to take a bit longer (not used to using such cruel instruments and the pain they cause will make it simply harder to operate in general)

Or, perhaps an enterprising individual wishes to use them for far more nefarious purposes, and is instead tormenting a "patient" with them. Each step taking longer means that it will cause more pain and thus more suffering for them.

plsleavemealon commented:
Anyway, I need to go get some sleep as its quite late where I live. I'm happy to keep discussing this though and make any further adjustments as required, if you feel they are necessary.

plsleavemealon commented:
@necromanceranne Is my view of the job flow in medical correct?

Also I should note that in the code for cruel tools they do specifically have a note saying that they are intended to be slower when used on live patients, but that is not functionally the case. Which is the other reason why I had the malus being applied more broadly before.

sippykot commented:
> First it causes morbid tools (any tool with the CRUEL_IMPLEMENT tag) to work slower when used on living targets for non-morbid surgeries, specifically, roughly 30% slower.
This might be cool in theory but in practice it will just be annoying (coroner has to have a 2nd set of tools to do normal doctor stuff)

plsleavemealon commented:
> > First it causes morbid tools (any tool with the CRUEL_IMPLEMENT tag) to work slower when used on living targets for non-morbid surgeries, specifically, roughly 30% slower.
> 
> This might be cool in theory but in practice it will just be annoying (coroner has to have a 2nd set of tools to do normal doctor stuff)

TBF, they shouldn't be doing normal doctor stuff to begin with

plsleavemealon commented:
Well I guess I have to ask, should cruel implements just function as normal tools then? What really is the point of them?

plsleavemealon committed:
more tweaks.

plsleavemealon committed:
more tweaking

accidentally pushed some things I was testing out previously.

Bibarra commented:
Are coroners better able to harvest body parts and organs with this change? Seems like a neat niche for them to be better at than a MD and gives them something to do outside of being a discount MD once they do their experiments.

plsleavemealon commented:
> Are coroners better able to harvest body parts and organs with this change? Seems like a neat niche for them to be better at than a MD and gives them something to do outside of being a discount MD once they do their experiments.

Yes, actually they already are faster at harvesting body parts and organs, as amputation surgery and organ manipulation surgery are both considered morbid surgeries.

plsleavemealon committed:
Merge branch 'master' into morbid-surgery-expansion

plsleavemealon commented:
> Oh that's an easy one. Sometimes I forget to update my code comments when I walk back a decision, so that's actually a mistake on my part. :U

Alrighty, then if its alright with you, I'm going to edit that comment out to something different.

"This object, when used for surgery, is causes a lot more pain for the patient, and are more efficient in a morbid users hands"

Is what I'll use in its place

> The Morbid surgeries exist because they're flavourful and subtly encourage the coroner to seek people out to perform them on people. If this was extended to, say, the experimental surgeries, this wouldn't be so much of a problem. But if it's literally tend wounds, thats when we have hit a snag, because that's the most commonly performed surgery in the game and also the bread and butter of medical doctors. I don't want to see coroners taking away their spotlight in that area.

You mentioned potentially expanding it to work with experimental surgeries. Would you like me to do that? It would be very easy for me to add those flags to those surgeries. And honestly a few of them absolutely should have it applied (looking at you necrotic revival).

Also, I could see if there is a way to add a temporary "CRUEL_IMPLEMENTS" tag to the surgery step where you are using an open hand to the "apply bioware" step (aka the part where you "thread veins" or "fold brains" step). This would make those steps cause much more pain in the patient if they are conscious during it....and frankly those steps absolutely should not be done on a patient without anesthetic lmao.

plsleavemealon committed:
removes all speed slowdown from the tools

Keeps only the pain increase they cause. Also tweaks the comment in the defines section to better reflect the nature of these tools.

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon committed:
Update obj_flags.dm

plsleavemealon commented:
Doing a lot of testing on how the surgery mood is applied and I have found some odd things that confuse me.

It seems that surgery mood only updates when you use a scalpel or cutting implement. This is true even for intentionally failing, and applies to _vanilla_ code. IE without any of my changes whatsoever. I have absolutely no idea why this is. Unfortunately, since this appears to be vanilla functionality this also extends to my own changes here.

So what this means is that cruel tools will only apply the surgical mood debuff for the cutting steps, IE scalpel or saw steps. Humorously, this also means that if you started using cruel tools, and then swapped to normal tools mid surgery for a cutting step, the mood debuff would be swapped to the surgical success mood debuff.

Example:
Organ manipulation surgery:
Cruel scalpel -> surgery failure mood
normal retractor -> mood unchanged
normal hemostat -> mood unchanged
normal scalpel -> surgery success mood overrides surgery failure
cruel cautery -> mood unchanged

The reverse is also true. So:
Normal scalpel -> surgery mood success
cruel retractor -> mood unchanged
cruel hemostat -> mood unchanged
cruel scalpel -> surgery failure mood overrides surgery success
normal cautery -> mood unchanged

plsleavemealon committed:
Update obj_flags.dm

plsleavemealon committed:
experimental surgery tweaks

gives experimental surgery the SURGERY_MORBID_CURIOSITY tag and also makes it so the manipulation step causes a bad mood without anesthesia

plsleavemealon committed:
Merge branch 'morbid-surgery-expansion' of https://github.com/plsleavemealon/tgstation into morbid-surgery-expansion

necromanceranne commented:
sorry that comment had some weird left over text from my earlier comment how weird, I'll give it a look in a bit.

plsleavemealon commented on code/modules/surgery/surgery_step.dm:
```diff
@@ -142,7 +142,10 @@
 		if((prob(100-fail_prob) || (iscyborg(user) && !silicons_obey_prob)) && !try_to_fail)
 			if(success(user, target, target_zone, tool, surgery))
-				update_surgery_mood(target, SURGERY_STATE_SUCCESS)
+				if((tool && tool.item_flags & CRUEL_IMPLEMENT) || (accept_hand && surgery.surgery_flags & SURGERY_MORBID_CURIOSITY))
```
Its honestly funny how easy it is to remove, there is even a comment about it in the code regarding it. And I tested this locally too to make sure it functioned like that. The moment you get any form of numbing agent, even if you aren't being operated on anymore it will go away and will not return.

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon committed:
Update surgery_step.dm

plsleavemealon commented:
@necromanceranne Any other changes?

plsleavemealon commented:
> I think this is plenty enough, sorry for all the runaround. Make sure to fix up your changelog and PR body and undraft it when you're ready for merge.

No worries! Designing stuff for a game isn't easy, and it can be hard to identify what the proper vision is for aspects of the game. I'm just glad we have been able to talk through things to figure out where best to go!

Anyway, I'll go ahead and run a few more tests since we have this all hammered out now just to make sure everything is kosher and then I'll get her undrafted.
